sdks/python/apache_beam/io/gcp/pubsub.py

```diff
@@ -36,16 +38,122 @@
 from apache_beam.transforms import window
 from apache_beam.transforms.display import DisplayDataItem
 
-__all__ = ['ReadStringsFromPubSub', 'WriteStringsToPubSub']
+__all__ = ['PubsubMessage', 'ReadMessagesFromPubSub', 'ReadStringsFromPubSub',
+           'WriteStringsToPubSub']
+
+
+class PubsubMessage(object):
+  """Represents a message from Cloud Pub/Sub.
+
+  This interface is experimental. No backwards compatibility guarantees.
+  """
+
+  def __init__(self, payload, message_id, attributes, publish_time):
+    """Constructs a message.
+
+    This interface is experimental. No backwards compatibility guarantees.
+
+    See also:
+      https://cloud.google.com/pubsub/docs/reference/rest/v1/PubsubMessage
+
+    Attributes:
+      payload: (str) Message payload, unencoded.
+      message_id: (str) Server assigned message ID. Unique within a topic.
+      attributes: (dict) Map of string to string.
+      publish_time: (str) Server assigned timestamp of when the message was
+        published.
+    """
+    self.payload = payload
+    self.message_id = message_id
+    self.attributes = attributes
+    self.publish_time = publish_time
+
+  def __eq__(self, other):
+    return isinstance(other, PubsubMessage) and (
+        self.payload == other.payload and
+        self.message_id == other.message_id and
+        self.attributes == other.attributes and
+        self.publish_time == other.publish_time)
+
+  def __repr__(self):
+    return '%s, %s, %s, %s' % (self.payload, self.message_id,
```

Comment:
<!--new_thread; commit:6bfe8d93b2cb863f16655addcbb98b35d64ce365; resolved:0-->
Could you include the class name, i.e. `'PubsubMessage(%s, %s, %s, %s)'`, to be consistent with the rest of the project?

Reply:
Done.